azure/services/scalesets/scalesets.go

```diff
@@ -616,15 +616,25 @@ func (s *Service) getVirtualMachineScaleSetNetworkConfiguration(vmssSpec azure.S
 				},
 			}
 
-			ipconfig.Subnet = &compute.APIEntityReference{
-				ID: pointer.String(azure.SubnetID(s.Scope.SubscriptionID(), vmssSpec.VNetResourceGroup, vmssSpec.VNetName, n.SubnetName)),
```

Comment:
removed this because it was redundant (it's already being set above on line 614)